content/en/docs/plugin-guide/plugin-policy-engine/packages/spinnaker.deployment/tasks.before/deleteManifest.md

```diff
@@ -0,0 +1,73 @@
+This example policy will prevent deleteManifest tasks from running unless they provide a grace period of 30 seconds or more.
+
+{{< prism lang="rego" line-numbers="true" >}}
+package spinnaker.deployment.tasks.before.deleteManifest
+
+deny["A minimum 30 second grace period must be given when deleting a kubernetes manifest"] { 
+  input.deploy.options.gracePeriodSeconds<30
+}
+{{< /prism >}}
+
+## Keys
+
+| Key                                       | Type      | Description                                                                                                                               |
+| ----------------------------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------- |
+| `input.deploy.account`                    | `string`  | The spinnaker account being deployed to.                                                                                                  |
+| `input.deploy.credentials`                | `string`  | The credentials to use to access the account.                                                                                             |
+| `input.deploy.labelSelectors.empty`       | `boolean` |                                                                                                                                           |
+| `input.deploy.labelSelectors.notEmpty`    | `boolean` |                                                                                                                                           |
+| `input.deploy.location`                   | `string`  | The name of the namespace from which the manifest is being deleted.                                                                       |
+| `input.deploy.manifestName`               | `string`  | The name of the manifest being deleted.                                                                                                   |
+| `input.deploy.options.apiVersion`         | `string`  | The API version in which the manifest's kind is defined.                                                                                  |
+| `input.deploy.options.dryRun`             | `boolean` | If true then the manifest will not actually be deleted. if false it will be.                                                              |
+| `input.deploy.options.gracePeriodSeconds` | `number`  | How many seconds should the resource being deleted be given to shut down gracefully before being forcefully shut down.                    |
+| `input.deploy.options.kind`               | `string`  | What is the kind of manifest that is being deleted.                                                                                       |
+| `input.deploy.options.orphanDependents`   | `boolean` | When set, delete all resources managed by this resource as well (all pods owned by a replica set). When unset, this may orphan resources. |
```

Comment:
Good question... @stephenatwell ?

Reply:
@alexfornuto  no, there should probably be an 'e.g.' there though since that's an example of the resources managed by a particular resource type (Specifically a replica set).